docs/devguide/xtypes.rst

```diff
@@ -1042,6 +1042,15 @@ The operations used for this include the DynamicData operations named "set_*" fo
 They work similarly to the "get_*" operations that are described in section :ref:`xtypes--interpreting-data-samples-with-dynamicdata`.
 When populating the DynamicData of complex data types, use get_complex_value() (see :ref:`xtypes--reading-members-of-more-complex-types`) to navigate from DynamicData representing containing types to DynamicData representing contained types.
 
+Unions have the following semantics in the OpenDDS implementation of DynamicData.
+Unions start in an "empty" state meaning that no branch is selected.
+At the point of serialization, the middleware will treat an empty union according to the following procedure.
+If the empty union has a default branch, then the union is serialized as though the default branch is selected with a default value for the corresponding member.
+If the empty union does not have a default branch, then the union is serialized as though the discriminant was assigned a default value.
```

Comment:
also below.... the name of this thing in the IDL spec is the "union discriminator"

Reply:
Okay.  Then we are using discriminant incorrectly in other places.

Reply:
I'd like to clarify this since this isn't exactly what `DynamicDataImpl` is doing to serialize an empty union. In this case, the discriminator is considered first and takes the default value of the discriminator type. Then, if it selects a branch, which can be the default or a non-default member, that branch will take the default value of its type, and the serialized data has both the discriminator and that branch. If it doesn't select a branch, the serialized data contains only the discriminator value.